ee/clickhouse/models/property.py

```diff
@@ -240,41 +234,36 @@ def prop_filter_json_extract(
         )
     elif operator == "is_date_after":
         # introducing duplication in these branches now rather than refactor too early
+        assert isinstance(prop.value, str)
         prop_value_param_key = "v{}_{}".format(prepend, idx)
-        query = f"AND parseDateTimeBestEffortOrNull({property_expr}) > %({prop_value_param_key})s"
-
-        if (
-            prop.property_definition is not None
-            and prop.property_definition.format is not None
-            and prop.property_definition.format in [UNIX_TIMESTAMP, UNIX_TIMESTAMP_MILLISECONDS]
-        ):
-            # ClickHouse can only parse 9 or 10 digit unix timestamps.
-            # So we drop the fractional seconds from millisecond timestamps
-            # or from after decimal place in seconds timestamps
-            query = f"AND parseDateTimeBestEffortOrNull(substring({property_expr}, 1, 10)) > %({prop_value_param_key})s"
+
+        query = f"""AND arrayFirst(x -> isNotNull(x), [
+            parseDateTimeBestEffortOrNull({property_expr}),
+            parseDateTimeBestEffortOrNull(substring({property_expr}, 1, 10))
```

Comment:
As discussed, what about 9-digit... or 12-digit timestamps, so slightly earlier dates? Or is this over thinking it? We can also just really under-think it and only keep the first part... I'm not sure anymore 😅

Reply:
ClickHouse only [supports 9 and 10 digit unix timestamps](https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions/#parsedatetimebesteffort). I can add a test to call out the less than 9 digit case.

By truncating either timestamps in milliseconds or timestamps in seconds with a decimal part, we lose any parts of a second represented so our granularity for querying timestamps is by the second. Which feels good enough as a starting position